nixos/modules/services/security/bitwarden-directory-connector.nix

```diff
@@ -0,0 +1,346 @@
+          example = "ou=groups";
+        };
+        objectClass = mkOption {
+          type = types.str;
+          description = lib.mdDoc "A class that groups will have.";
+          example = "groupOfNames";
+        };
+        nameAttribute = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Attribute for a name of group.";
+          example = "cn";
+        };
+        filter = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Filter for groups.";
+          example = "(cn=sales)";
+        };
+      };
+    };
+
+    env = {
+      ldap = mkOption rec {
+        type = types.str;
+        description = "Auth for the LDAP, has value defined in {option}`ldap.pw_env";
+      };
+      bitwarden = mkOption rec {
+        type = types.str;
+        description = "Auth for Bitwarden, has BW_CLIENTID and BW_CLIENTSECRET";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups."${cfg.user}" = {};
+
+    users.users."${cfg.user}" = {
+      createHome = true;
+      isSystemUser = true;
+      home = "${cfg.directory}";
+      group = "${cfg.user}";
+      homeMode = "711";
+    };
+
+    systemd = {
+      timers.bitwarden_directory_connector = {
+        description = "Sync timer for Bitwarden Directory Connector";
+        wantedBy = ["timers.target"];
+        partOf = ["bitwarden_directory_connector.service"];
+        timerConfig = {
+          OnCalendar = cfg.sync.interval;
+          Unit = "bitwarden_directory_connector.service";
+          Persistent = true;
+        };
+      };
+
+      services.bitwarden_directory_connector = {
+        description = "Main process for Bitwarden Directory Connector";
+        wantedBy = ["multi-user.target"];
+        after = ["network-online.target"];
+        wants = [];
+
+        environment = {
+          BITWARDENCLI_CONNECTOR_APPDATA_DIR = cfg.directory;
+          BITWARDENCLI_CONNECTOR_PLAINTEXT_SECRETS = "true";
+        };
+
+        serviceConfig = {
+          Type = "oneshot";
+          User = "${cfg.user}";
+          Group = "${cfg.user}";
+          ExecStartPre = pkgs.writeShellScript "bitwarden_directory_connector-config" ''
+            # create the config file
+            ${cfg.package}/bin/${cfg.binary_name} data-file
+
+            ${cfg.package}/bin/${cfg.binary_name} config server ${cfg.domain}
+
+            # now login to set credentials
+            ${cfg.package}/bin/${cfg.binary_name} login
+
+            # set the ldap details
+            sed -i 's/"ldap": null/"ldap": ${sed_string ldap_data}/' ${cfg.directory}/data.json
+
+            # set the client id
+            orgID=$(echo $BW_CLIENTID | sed 's/organization\.//g')
+            sed -i "s/\"organizationId\": null/\"organizationId\": \"$orgID\"/" ${cfg.directory}/data.json
+
+            # and sync data
+            sed -i 's/"sync": null/"sync": ${sed_string sync_data}/' ${cfg.directory}/data.json
```

Comment:
I'm a bit leery of using `sed` on json data. `jq` is a much more robust tool for the job.

Reply:
Was able to remove almost all the sed's.   
Sits better with me as well.